### PR TITLE
remove `AllSubsystems` and `AllSubsystemsGen` types

### LINKED ISSUE #3773: [overseer] remove `AllSubsystems` and proc-macro `AllSubsystemGen`
Currently a remainder of the past, `AllSubsystems*` items are used in the `OverseerGen` pattern.

Thes should be removed ultimately. The ground work is already set via #3775 which introduces the `OverseerConnector` (`#overseer Connector`) type.

### PR DESCRIPTION
**Pull request by drahnr**
Second part of #3803 - will be rebased on-top of #3868 to end up at the same changeset as #3803

Closes #3773

### CODE REVIEW

File: node/network/statement-distribution/src/lib.rs [hunk 1]
```diff
@@ -105,7 +105,7 @@ const MAX_LARGE_STATEMENTS_PER_SENDER: usize = 20;
 
 /// The statement distribution subsystem.
 pub struct StatementDistribution {
-	/// Pointer to a keystore, which is required for determining this nodes validator index.
+	/// Pointer to a keystore, which is required for determining the nodes validator index.
```
Comment (rphmeier) [inline node/network/statement-distribution/src/lib.rs:108]:
```suggestion
	/// Pointer to a keystore, which is required for determining this node's validator index.
```

File: node/overseer/src/lib.rs [hunk 2]
```diff
@@ -385,328 +492,77 @@ pub struct Overseer<SupportsParachains> {
...
+		// into one `to_overseer` value.
+		metronome_metrics.channel_fill_level_snapshot(
+			subsystem_meters
+				.iter()
+				.cloned()
+				.filter_map(|x| x)
+				.map(|(name, ref meters)| (name, meters.read())),
+		);
+
+		async move { () }
```
Comment (rphmeier) [inline node/overseer/src/lib.rs:545]: `future::ready(())` is more readable to me

**Review by ordian — APPROVED** [review]

File: node/service/src/overseer.rs [hunk 3]
```diff
@@ -106,165 +109,6 @@ where
 	pub dispute_coordinator_config: DisputeCoordinatorConfig,
 }
 
-/// Create a default, unaltered set of subsystems.
-///
-/// A convenience for usage with malus, to avoid
-/// repetitive code across multiple behavior strain implementations.
-pub fn create_default_subsystems<'a, Spawner, RuntimeClient>(
```
Comment (rphmeier) [inline node/service/src/overseer.rs:113]: Was this moved into `malus`?

Comment (drahnr) [reply node/service/src/overseer.rs:113]: There are no more `AllSubsystems` instances, so `fn prepared_overseer_builder` takes its place instead.

Comment (rphmeier) [reply node/service/src/overseer.rs:113]: I see, `malus` is now using `prepared_overseer_builder`.

**Review by rphmeier — APPROVED** [review]
Looks good, although I didn't review the macro code closely.

File: node/service/src/overseer.rs (hunk 3 above)
Comment (drahnr) [reply node/service/src/overseer.rs:113]: And `fn dummy_overseer_builder` respectively.

**Comment by drahnr** [comment]
bot merge

**Pull request merged**